### PR TITLE
Feature: My Account: Search Implemented: UserActivity

### PR DESCRIPTION
**Pull request by BobbySandhu**
- [x] I have reviewed the pull request myself. 
- [x] I confirm that code style is correct and follows CrownStack code style guide.
- [x] I have checked the code after running on Simulator/emulator/Device (which ever applicable).
- [x] I am convinced that I have not added new bugs. 
- [x] The UI behaviour is correct.

### CODE REVIEW

**Comment by ashutoshsingh33** [comment]
@aashishdhawan  Please review .

File: app/src/main/java/com/eqdepot/activities/AddUserActivity.java [hunk 1]
```diff
@@ -36,9 +60,90 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
...
+        String firstName = editTextFirstName.getText().toString();
+        String lastName = editTextLastName.getText().toString();
+        String emailAddress = editTextEmail.getText().toString();
+        String phoneNumber = editTextPhoneNumber.getText().toString();
+
+        if (validate(firstName, lastName, emailAddress, phoneNumber))
+            processAddUser(firstName, lastName, emailAddress, phoneNumber);
+    }
+
+    private void processAddUser(String firstName, String lastName, String email, String phoneNumber) {
```
Comment (aashishdhawan) [inline app/src/main/java/com/eqdepot/activities/AddUserActivity.java]: the name should be OK. `processAddUser` is not OK.

File: app/src/main/java/com/eqdepot/activities/AddUserActivity.java [hunk 2]
```diff
@@ -36,9 +60,90 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
...
+    }
+
     private void setFont() {
         FontHelper.setFontFace(FontHelper.FontType.FONT_MEDIUM, textViewFirstName, textViewLastName, textViewEmailAddress,
                 textViewPhoneNumber);
         FontHelper.setFontFace(FontHelper.FontType.FONT_BOLD, buttonAddUser);
     }
-}
+
+    private void confirmPopup(String message) {
```
Comment (aashishdhawan) [inline app/src/main/java/com/eqdepot/activities/AddUserActivity.java]: Again name should be OK. Something like `showConfirmPopup`

File: app/src/main/java/com/eqdepot/adapters/UsersAdapter.java [hunk 3]
```diff
@@ -42,17 +45,28 @@ public AddUserViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
...
+        holder.bindData(mAllUserList.get(position));
     }
 
     @Override
     public int getItemCount() {
-        return mArrayList.size();
+        return mAllUserList.size();
+    }
+
+    public void getFilteredItem(String args) {
```
Comment (aashishdhawan) [inline app/src/main/java/com/eqdepot/adapters/UsersAdapter.java]: Please change the parameter name `args`, does not make any sense.

File: app/src/main/java/com/eqdepot/adapters/UsersAdapter.java [hunk 4]
```diff
@@ -42,17 +45,28 @@ public AddUserViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
...
+        mAllUserList.clear();
```
Comment (aashishdhawan) [inline app/src/main/java/com/eqdepot/adapters/UsersAdapter.java]: Please pass `mData` as a parameter in this function. A function should have very less dependency on outside world. Also name `mData` does not make any sense to me. Can we make it better?

File: app/src/main/java/com/eqdepot/adapters/UsersAdapter.java [hunk 5]
```diff
@@ -42,17 +45,28 @@ public AddUserViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
...
+        if (!args.equals("")) {
+            for (int i = 0; i < mData.size(); i++) {
+                if ((mData.get(i).getFirstName() + " " + mData.get(i).getLastName()).toLowerCase().contains(args.toLowerCase()))
```
Comment (aashishdhawan) [inline app/src/main/java/com/eqdepot/adapters/UsersAdapter.java]: too complex to understand. Please break this 
```java
String name = mData.get(i).getFirstName() + " " + mData.get(i).getLastName();
if (name.toLowerCase().contains(args.toLowerCase()) {
}
```

**Pull request merged**